src/parser/ast.c

```diff
@@ -172,37 +172,6 @@ static AST_Validation _Validate_MATCH_Clause(const AST *ast, char **reason) {
   }
 
   TrieMap_Free(edgeAliases, TrieMap_NOP_CB);
-
-  /* Verify that no alias appears in multiple independent patterns.
-   * TODO We should introduce support for this when possible. */
-  int patternCount = Vector_Size(ast->matchNode->patterns);
-  if (Vector_Size(ast->matchNode->patterns) > 1) {
-    TrieMap *reused_entities = NewTrieMap();
-    Vector *pattern;
-    AST_GraphEntity *elem;
-
-    int pattern_ids[patternCount];
-    for (int i = 0; i < patternCount; i ++) {
-      pattern_ids[i] = i;
-      Vector_Get(ast->matchNode->patterns, i, &pattern);
-      for (int j = 0; j < Vector_Size(pattern); j ++) {
-        Vector_Get(pattern, j, &elem);
-        char *alias = elem->alias;
-        if (!alias) continue;
-        void *val = TrieMap_Find(reused_entities, alias, strlen(alias));
-        // If this alias has been used in a previous pattern, emit an error.
-        if (val != TRIEMAP_NOTFOUND && *(int*)val != i) {
-          asprintf(reason, "Alias '%s' reused. Entities with the same alias may not be referenced in multiple patterns.", alias);
```

Comment:
It's so great to have this restriction removed! \o/

Reply:
Indeed it is, took us awhile but we're finally made it!